book/src/04_4_interrupts.md

```diff
@@ -1,5 +1,72 @@
 # Interrupts
 
-TODO link isr exercise
-- button press -> log message
+The goal of this exercise is to handle a button interrupt, so that the words "button pushed" get logged, if the `BOOT` button is pushed. 
+This exercise involves working with C bindings to the esp-idf-sys and other unsafe operations, as well as non-typical rust documentation. In a first step we will go line by line to build this interrupt handler. 
+
+You can find a skeleton code for this exercise in `advanced/button-interrupt/exercise/src/main.rs.`
+You can find the solution for this exercise in advanced/button-interrupt/solutuion/src/main.rs.
```

Comment:
```suggestion
You can find the solution for this exercise in `advanced/button-interrupt/solution/src/main.rs`
```

Is there a way to hyper-link to the source code? Maybe...:

```suggestion
You can find the solution for this exercise in [`advanced/button-interrupt/solution/src/main.rs`](https://github.com/ferrous-systems/espressif-trainings/tree/main/advanced/button-interrupt/solution)
```

I don't know if people are more likely to be reading online, or using a local clone of the repo with `mdbook serve` running. If the latter, you would need a relative link somehow.